docs/source/format/CanonicalExtensions.rst

```diff
@@ -251,6 +251,17 @@ Variable shape tensor
    Values inside each **data** tensor element are stored in row-major/C-contiguous
    order according to the corresponding **shape**.
 
+UUID
+====
+
+* Extension name: `arrow.uuid`.
+
+* The storage type of the extension is ``FixedSizeBinary`` with a length of 16 bytes.
```

Comment:
Isn't `FixedSizeBinary` a logical type? Is it specified somewhere what physical layout should be used? Unless I'm mistaken there are two options: fixed-size primitive layout (`FixedSizePrimitive<[u8; 16]>`) or fixed-size list layout (`FixedSizeList<u8>(16)`)?

Reply:
I assumed `FixedSizeBinary` maps to a single physical type (fixed size primitive). If this is ambiguous we should indeed specify (preferrably one) physical type.

Reply:
AFAIK we want to use a logical type here, since that are the concrete types that are actually used in the various specs like IPC and C Data Interface (the phsyical layouts are just described in the format docs, but it are the logical types that are listed in the flatbuffer Schema.fbs). 
While "storage type" might sound like it points to a physical layout, the docs also mention that extension types annotate any built-in logical type: https://arrow.apache.org/docs/dev/format/Columnar.html#extension-types

Sidenote: there is not really such a thing as a "physical type", but only physical _layout_. Next to the various layouts, we just have "types" on top of that, which are sometimes called "logical types", but we are not very consistent about that terminology ..